mate-panel/panel-xutils.c

```diff
@@ -100,14 +105,24 @@ panel_xutils_set_strut (GdkWindow        *gdk_window,
         break;
     }
 
+    area[0] = rect->x;
+    area[1] = rect->y;
+    area[2] = rect->width;
+    area[3] = rect->height;
```

Comment:
`rect` should be scaled to device pixels - https://gitlab.gnome.org/GNOME/gnome-panel/-/commit/e85b2d2138976ba2c315ab438a50644e795a3fb6.

Reply:
Thx, this helps.